### ISSUE TITLE
Issues with new sidebar menu

### ISSUE THREAD

**Issue by eksperimental** [elixir-lang/ex_doc#232]
I have found a few glitches, I will list them:

1. ~~When visiting a module page, the link to this module page in the menu shouldn't link to a new page, but to the top of the page itself (we probably should update the address in the address bar to be /Module.html and not /Module.html#)
Example: http://elixir-lang.org/docs/master/elixir/Base.html clicking on Base (on the menu) shouldn' t reload the page).~~

2. ~~The clear icon (red X) to reset the search filed is gone. I find it very useful.~~

3. Input text is lost when Switching between " Modules, Exceptions and Protocols"  once you searched something, it will never shot the results in the other sections, ~~and the loading icon hangs there endlessly.~~
if something is found in modules, you go to exceptions and no results are shown, and once back to modules, it will show nothing. you can try this with the term "into" swapping back and forth between **Modules** to **Protocols**

4. ~~Clicking any module in exception when the page is loaded won't display the Exception/Protocol tab by default in the menu. Example: http://elixir-lang.org/docs/master/elixir/MatchError.html#message/1~~
 
5. ~~~Hitting the clear button in the search box, should focus the cursor in the search box.~~~

6. ~~**Misleading Behaviour of the Clear button**: this is a tricky situation with the X button, you have to hit the actual "X" for it to actually work. If you hit the button, but you miss the "X" it looses focus on the search box, and it gives the impression that the clear button has been hit, but it hasn't. You need to hit the actual "X" for it to work. (Browser: latest Chromium)~~

7. ~~~**Internal links do not work** This is new, probably introduced by the latest chances. When a new page is loaded, by follwing any link or when clicking on a function of the different module than the current page, it will take you to the module page, but not the the internal link of that function, even though you can see the proper link int the address bar.~~~

**Comment by josevalim** [comment]
Thanks for reporting @eksperimental! We would love if you can you help with fixing some of those too!

**Comment by milmazz** [comment]
@eksperimental Thanks for your report, maybe we can solve all this issues if we do the following:

1. Right now the `fillSidebarWithNodes` function is in charge of the creation of the `#full_list` children, so, through this function I think is feasible to control the behaviour that you mentioned.
2. Maybe there is some missing CSS rules or HTML layout.
3. Maybe this can be fix if we add something like:

```javascript
sidebarNav.on('click', '#modules_list,#exceptions_list,#protocols_list', function ()  {
  $('#search input').val('');
}
```

**Comment by eksperimental** [comment]
you are welcome @josevalim, I don't think I will time to contribute for now ;-(  , but maybe I manage to do a little bit here and there.

**Comment by eksperimental** [comment]
I have updated the list with a 4th issue:
> 4. Clicking any module in exception when the page is loaded won't display the Exception/Protocol tab by default in the menu. Example: http://elixir-lang.org/docs/master/elixir/MatchError.html#message/1

**Comment by eksperimental** [comment]
@milmazz i realized now when you hit the clear (X) , the focus on the search box is lost. and you need to click again inside the box to start typeing again.

**Comment by milmazz** [comment]
@eksperimental Ok, I'll push a fix for that asap.

Can you please update the list of issues remaining on this ticket?

**Comment by eksperimental** [comment]
@milmazz thanks a lot.
Next week hopefully I will be able to catch up on ex_doc with the latest changes. for now I will just report whatever issues I come across
I will update that list now.

**Comment by eksperimental** [comment]
@milmazz I have updated the list. Please let me know if I missed something.

**Comment by eksperimental** [comment]
@milmazz  I have added 2 new issues that i found ( no. 6 and 7). thank you for your work ;-D

**Comment by milmazz** [comment]
@eksperimental Thanks for updating the list! Please, can you tell me how to reproduce 7?

**Comment by eksperimental** [comment]
@milmazz it is not happening in the live docs, but it is happening in my local copy, which was just pulled minutes before i reported the issue.
http://elixir-lang.org/docs/master/elixir/Agent.html#start/4
just directly visit any link to a function, 
if you still cannot reproduce it, please let me know, I will check if it has to do with running on elixir-dev

**Comment by josevalim** [comment]
#7 is fixed. @milmazz can we fix #3 and #6 and get a new release out? :D

**Comment by eksperimental** [comment]
I think the margins in the content area. are too narrow.
it looks too packed against the side bar.
the rigth margin could also be wider, what do you guys think?

**Comment by josevalim** [comment]
@eksperimental let's focus on getting the functionality right for now. There is a new layout coming and we haven't released in ages.

**Comment by josevalim** [comment]
I have just fixed #6 too. Now taking a look at #3. :D

**Comment by josevalim** [comment]
Ok, I can't fix #3 because I am unsure how to trigger only the search when changing the tab. I will leave it up for @milmazz as he definitely has more knowledge than me on this!

**Comment by eksperimental** [comment]
regarding no. 3
Just to be clear, because the way I described the issue is a bit of a counter intuitive the behaviour..
What i expect is when we type "ab" while in modules, we click on any another tab, and it should show us the results for "ab" in that tab. and not keep different states for different tabs.

**Comment by josevalim** [comment]
Ok, I have fixed #3 in master! :D

Please give it a try a couple more tries, I will update the master version, and we should release something soon!™

**Comment by eksperimental** [comment]
It looks great,
thank you guys for the great work
